spec-linter/spec_linter.go

```diff
@@ -0,0 +1,190 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/fs"
+	"maps"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var (
+	digitalDatasheetDir = flag.String("datasheet-dir", "", "Locational of digital-datasheets repo on local machine")
+)
+
+func main() {
+	flag.Parse()
+
+	fileList := []string{}
+	err := filepath.Walk(*digitalDatasheetDir, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Skip .git and part-spec-proto directories
```

Comment:
The code below essentially assumes that the files it is looking at is in the `part-spec` directory (as per the check which requires the $id to have a prefix that contains `/part-spec/`. We could get rid of this tricky check to ignore specific files by just calling Walk on `filepath.Join(*digitalDatasheetDir, "part-spec")` to directly get the directory we actually care about.

Reply:
that's a great idea!